### ISSUE TITLE
Compilation fails with latest emscripten 

### ISSUE THREAD

**Issue by djdeath** [protocolbuffers/protobuf#4122]
In the src/google/protobuf/stubs/port.h file, there is this bit of defines : 

```
#if defined(__clang__) && defined(__has_cpp_attribute) \
    && !defined(GOOGLE_PROTOBUF_OS_APPLE)
# if defined(GOOGLE_PROTOBUF_OS_NACL) || defined(EMSCRIPTEN) || \
     __has_cpp_attribute(clang::fallthrough)
#  define GOOGLE_FALLTHROUGH_INTENDED [[clang::fallthrough]]
# endif
#elif defined(__GNUC__) && __GNUC__ > 6
# define GOOGLE_FALLTHROUGH_INTENDED [[gnu::fallthrough]]
#endif

#ifndef GOOGLE_FALLTHROUGH_INTENDED
# define GOOGLE_FALLTHROUGH_INTENDED
#endif

```

Compilation using the latest version of emscripten (1.37.27), compiling protobuf fails because the ```[[clang::fallthrough]]``` isn't recognized : 

```
emsdk-portable/emscripten/1.37.27/em++ -s WASM=1 -O2 -Iprotoc@sha -I. -I.. -I../src -Xclang -fcolor-diagnostics -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wnon-virtual-dtor -O0 -g -fPIC -DHAVE_PTHREAD -Wno-sign-compare -Wno-unused-parameter -Wno-ignored-qualifiers -Wno-unused-function -MMD -MQ 'protoc@sha/src_google_protobuf_compiler_cpp_cpp_map_field.cc.o' -MF 'protoc@sha/src_google_protobuf_compiler_cpp_cpp_map_field.cc.o.d' -o 'protoc@sha/src_google_protobuf_compiler_cpp_cpp_map_field.cc.o' -c ../src/google/protobuf/compiler/cpp/cpp_map_field.cc
In file included from ../src/google/protobuf/compiler/cpp/cpp_map_field.cc:32:
In file included from ../src/google/protobuf/compiler/cpp/cpp_helpers.h:42:
In file included from ../src/google/protobuf/descriptor.pb.h:25:
In file included from ../src/google/protobuf/generated_message_table_driven.h:35:
../src/google/protobuf/map_entry_lite.h:204:11: error: expected expression
          GOOGLE_FALLTHROUGH_INTENDED;
          ^
../src/google/protobuf/stubs/port.h:244:39: note: expanded from macro 'GOOGLE_FALLTHROUGH_INTENDED'
#  define GOOGLE_FALLTHROUGH_INTENDED [[clang::fallthrough]]
                                      ^
1 error generated.
ERROR:root:compiler frontend failed to generate LLVM bitcode, halting
```
I think the define above should probably drop the defined(EMSCRIPTEN). If emscripten allows ```[[clang::fallthrough]]``` it will probably expose it through ```__has_cpp_attribute``` and if not, then compilation won't fail.

**Comment by xfxyjwf** [comment]
We don't test protos with emscripten but we are happy to accept pull requests to make it work there.

**Comment by djdeath** [comment]
I think it's fixed by dfaf1aac309c1b65679bb28fb593303ae3e8d170.
